src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h

```diff
@@ -1743,12 +1743,9 @@ namespace UiaOperationAbstraction
 
         VARIANT get() const;
 
-        // It would be nice if this didn't have to be told what the unsupported attribute value is,
-        // but we don't have an IUIAutomation object handy in this library so it does.
-        UiaBool IsNotSupported(_In_ const IUnknown* unsupportedAttributeValue) const;
+        UiaBool IsNotSupported() const;

```

Comment:
Hmm, I think you're right it doesn't make sense to need to take an `IUnknown` as a parameter when we do already have `g_automation`. Probably this is left over from a _very_ old version of the code. Let me think about it some more to make sure we're not missing anything.

Reply:
OK, I've thought about this some. Yeah, this parameter here is absolutely something left over from a while ago. Good catch!

Personally, I think it's fine to in this change remove the parameter. This UiaOperationAbstraction is still super young and pre-versioned, so breaking API compatibility doesn't really bother me too much. Plus, this breaking compatibility would just result in a build break, so it shouldn't cause unexpected behavior if anyone were to update their UiaOperationAbstraction dependency. Does that seem reasonable from your perspective?

@pengyuanjin @zhayuli @mavangur any particular thoughts about this?

Reply:
I'm certainly good with keeping the change. It has confused me as to why it was there for a very long time :)

Reply:
This was indeed a remnant from long ago - one of the first iterations of the library.

Originally, the idea was that this helper wouldn't maintain any state.

(In fact, at an even earlier point, the idea was to try to make this a header-only library; though that's orthogonal to whether or not you maintain static state, I believe there are still some remnants of that approach in other parts of the code that you might find, but I digress...)

Static libraries that do that are finnicky... What happens when you've got two clients in the same module (exe/dll) who might want to configure that static state differently? Maybe one uses Win7 IUIAutomation and the other the newer one... If you allow static state to be updated, then you've got to think about thread safety - which either leads to a "please don't do this concurrently" code comment which someone will not see and almost certainly eventually lead to a crash or a perf degradation due to too much locking. 

So, since we didn't want any static state, we wanted to avoid maintaining a `IUIAutomation` objects internally... As such, the only way to implement this would've been to pass the object at call time (either automation or the punk).

However, over time and once we started using this in real code, that proved overly limiting as more and more scenarios showed up where having that interface around would've been nifty and led to better code and interfaces. Meanwhile, there didn't seem to be a _real_ scenario where a single module (mostly DLLs) would want to initialize this differently. So we decided to adjust that original thinking and added the init function that accepts automation. The fact that it's "init" should hopefully also avert anyone from modifying it, mitigating the concurrency concerns. 

(I do think we left a warning comment somewhere about this potentially leading to surprising results if two different clients in the module run init?)

However, once we added that, we simply never went back to update these methods.

So +1 to this improvement. 

Hope the historic background helps.